### DISCUSSION TITLE
[feat]: Add embedding model Engine "LiteLLM"

### DISCUSSION THREAD

**Discussion by blegry** [open-webui/open-webui#7722] [General]
# Feature Request

## Description

To centralized all OWUI transactions and manage tokens, I need new feature for embedding model target.

## Hope

Can you add another target for embedding ? Actually, there is "only" ollama and openAI possibility. 
It should be wonderfull if LiteLLM  was integrated as new embedding model engine.

## Alternative

Actually, there is no alternative except directly pass through Ollama target. 

## Additional context

Image below from settings/documents/general settings:

![image](https://github.com/user-attachments/assets/9dc552b5-af4a-4f87-950a-59b5c99e80a8)

## Thanks

Thanks a lot for your jobs. very appreciate.

**Comment by flefevre** [comment]
I do agree: Litellm should be used both for emebdding and inference models. it will align the pratices

**Comment by tjbck** [comment]
OpenAI is LiteLLM.

**Reply by flefevre** [reply]
OK, thanks for this usefull precision.
Here some suggestions for updating the User Interface:

- perhaps it could be good to have the same widget that list available models such as the one defined in the Admin>Model Dashboard.
- The icon "download, should not be present if Openai mode is selected since it is only available for Ollama models : download model should not be managed at two places
- perhaps when we list available models from Litellm we could filter with the meta-data `model_info > mode "embedding" but i do not know if it is provided all time.


```
 - model_name: text-embedding-ada-002
    litellm_params: 
      model: azure/azure-embedding-model
      api_key: os.environ/AZURE_API_KEY
      api_base: https://openai-gpt-4-test-v-1.openai.azure.com/
      api_version: "2023-05-15"
    model_info:
      mode: embedding
      base_model: text-embedding-ada-002
``

**Comment by chourete** [comment]
please add Voyage model, its fast

**Comment by flefevre** [comment]
We can close the discussion since litellm is enable for embedding
